tools/astarte_import/lib/astarte/import.ex

```diff
@@ -0,0 +1,136 @@
+  end
+
+  defp xml_event(
+         {:startElement, _uri, _l_name, {_prefix, 'interfaces'}, _attributes},
+         _loc,
+         state
+       ) do
+    state
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'interface'}, attributes}, _loc, state) do
+    with {:ok, name} <- get_attribute(attributes, 'name'),
+         {:ok, major_string} <- get_attribute(attributes, 'version_major'),
+         {major, ""} <- Integer.parse(major_string),
+         {:ok, minor_string} <- get_attribute(attributes, 'version_minor'),
+         {minor, ""} <- Integer.parse(minor_string) do
+      %State{state | interface: {name, major, minor}}
+    else
+      _any ->
+        throw({:error, :invalid_interface})
+    end
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'values'}, attributes}, _loc, state) do
+    {:ok, path} = get_attribute(attributes, 'path')
+
+    %State{state | path: path}
+  end
+
+  defp xml_event({:startElement, _uri, _l_name, {_prefix, 'value'}, attributes}, _loc, state) do
+    {:ok, timestamp} = get_attribute(attributes, 'timestamp')
+    {:ok, datetime, 0} = timestamp |> DateTime.from_iso8601()
+
+    %State{state | timestamp: datetime}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'value'}}, _loc, state) do
+    %State{state | timestamp: nil}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'values'}}, _loc, state) do
+    %State{state | path: nil}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'interface'}}, _loc, state) do
+    %State{state | interface: nil}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'interfaces'}}, _loc, state) do
+    %State{state | interface: nil}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'device'}}, _loc, state) do
+    %State{state | device_id: nil}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'devices'}}, _loc, state) do
+    %State{state | device_id: nil}
+  end
+
+  defp xml_event({:endElement, _uri, _l_name, {_prefix, 'astarte'}}, _loc, state) do
+    %State{state | device_id: nil}
+  end
+
+  defp xml_event({:characters, chars}, _loc, %State{insert_value_fun: fun} = state) do
+    fun.(state, chars)
+  end
+
+  defp xml_event({:ignorableWhitespace, _whitespace}, _location, state) do
+    state
+  end
+
+  defp xml_event(:startDocument, _location, state) do
+    state
+  end
+
+  defp xml_event(:endDocument, _location, state) do
+    state
+  end
+
+  defp get_attribute(attributes, attribute_name) do
+    attribute_tuple =
+      Enum.find(attributes, fn attribute ->
+        case attribute do
+          {_uri, _prefix, ^attribute_name, _attribute_value} ->
+            true
+
+          _ ->
+            false
+        end
+      end)
+
+    with {_uri, _prefix, _attribute_name, attribute_value} <- attribute_tuple do
+      {:ok, to_string(attribute_value)}
+    else
+      nil ->
+        {:error, :missing_attribute, attribute_name}
```

Comment:
Wrap the "reason" in a tuple (`{:error, {:missing_attribute, attribute_name}}`) so that we're consistent with Elixir's APIs